packages/backend-common/src/database/SingleConnection.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Knex from 'knex';
+import { Config } from '@backstage/config';
+import { createDatabaseClient, ensureDatabaseExists } from './connection';
+import { PluginDatabaseClientFactory } from './types';
+
+/**
+ * Implements a Database Manager which will automatically create new databases for plugins when requested. All
+ * requested databases are created with the credentials provided.
+ */
+export class SingleConnectionManager {
+  private readonly config: Config;
+
+  constructor(config: Config) {
```

Comment:
Might be good to use a consistent pattern with a static `fromConfig` here. Similar to how `SingleDiscoveryHost.fromConfig`.

Reply:
I think my latest push normalises this, thanks!